runtime/src/accounts_db.rs

```diff
@@ -1121,6 +1121,29 @@ impl AccountsDB {
                 }
             }
         }
+
+        let mut counts = HashMap::new();
+        for slot_list in accounts_index.account_maps.values() {
+            for (_slot, account_entry) in slot_list.read().unwrap().iter() {
+                *counts.entry(account_entry.id).or_insert(0) += 1;
+            }
+        }
+        for slot_stores in storage.0.values() {
+            for (id, store) in slot_stores {
+                if let Some(count) = counts.get(&id) {
+                    trace!(
+                        "id: {} setting count: {} cur: {}",
+                        id,
+                        count,
+                        store.count_and_status.read().unwrap().0
+                    );
+                    store.count_and_status.write().unwrap().0 = *count;
```

Comment:
This discards the original value (=deserialized from snapshot) of `count` of `count_and_status` by unconditionally overwriting it.